src/Model/Config/Communication.php

```diff
@@ -43,16 +43,29 @@ public function getParameters(): array
             'currency-code'               => $this->view->getActCurrency()->name,
             'currency-fields'             => $this->getAdditionalCurrencyFields(),
             'currency-country-code'       => $this->getLocale($this->view->getActiveLangAbbr()),
-            'add-params'                  => $this->useForCategories() ? $this->getCategoryPath($category) : 'cl=search_result',
             'search-immediate'            => $this->isSearch() || $this->useForCategories() ? 'true' : 'false',
             'keep-url-params'             => 'cl',
             'only-search-params'          => 'true',
             'use-browser-history'         => 'true',
         ];
 
+        $params = $this->prepareParameters($params, $category);
+
         return array_filter($this->mergeParameters($params, $this->getAdditionalParameters()));
     }
 
+    protected function prepareParameters(array $params, ?Category $category): array
+    {
+        if ($this->getConfig('ffApiVersion') === 'ng') {
+            $params['category-page'] = $this->useForCategories() ? $this->getCategoryPath($category) : null;
+            $params['add-params']    = $this->isSearch() ? 'cl=search_result' : null;
+        } else {
+            $params['add-params'] = $this->useForCategories() ? $this->getCategoryPath($category) : 'cl=search_result';
```

Comment:
@a-laurowski can you take a look at this condition, the build fails because of `else` but I think it should be there because in case I remove it the `add-params` will be filled twice if api version is `ng` and this is not necessary to call useForCategories function and then check another condition in `if`. What do you think, should we keep this else or remove it?